rayon-core/src/registry.rs

```diff
@@ -513,27 +540,35 @@ pub struct WorkerThread {
 // thread local variable. So it will remain valid at least until the
 // worker is fully unwound. Using an unsafe pointer avoids the need
 // for a RefCell<T> etc.
-thread_local! {
-    static WORKER_THREAD_STATE: Cell<*const WorkerThread> =
-        Cell::new(0 as *const WorkerThread)
-}
+#[thread_local]
```

Comment:
FWIW This can be used for profiling, but we can't use this in rustc because it doesn't work on Windows (and possibly some other platforms, I forget)

Reply:
We do use that for `thread_local!` on Windows right? It can't be too bugged. Also when are we getting a `thread_local!` which accepts constant initializers?

Reply:
No on Windows we use the system thread-local APIs. We tried using `#[thread_local]` in the past and it was buggy for a reason I don't remember